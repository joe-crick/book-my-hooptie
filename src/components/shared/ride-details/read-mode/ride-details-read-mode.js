import React from "react";
import styles from "./read-mode.less";
import { translate } from "react-i18next";

export const RideDetailsReadMode = ({ pickup, dropoff, dateTime, t: translate }) => (
  <div className={styles.container}>
    <div>
      <strong>{translate("pickup.label")}</strong> {pickup}
    </div>
    <div>
      <strong>{translate("dropoff.label")}</strong> {dropoff}
    </div>
    <div>
      <strong>{translate("booking.pickupDate")}</strong> {dateTime}
    </div>
  </div>
);

export default translate()(RideDetailsReadMode);
