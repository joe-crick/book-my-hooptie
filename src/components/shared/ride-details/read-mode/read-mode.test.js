import React from "react";
import { RideDetailsReadMode } from "./ride-details-read-mode";
import renderer from "react-test-renderer";
import { i18nMock } from "test/test-utils";

describe("Header", () => {
  it("should render", () => {
    const component = renderer.create(
      <RideDetailsReadMode
        pickup="pickup"
        dropoff="dropoff"
        dateTime="date-time"
        t={i18nMock}
      />
    );
    let readMode = component.toJSON();
    expect(readMode).toMatchSnapshot();
  });
});
