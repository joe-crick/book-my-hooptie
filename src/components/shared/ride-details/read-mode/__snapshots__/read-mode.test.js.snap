// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<div
  className={undefined}
>
  <div>
    <strong>
      pickup.label
    </strong>
     
    pickup
  </div>
  <div>
    <strong>
      dropoff.label
    </strong>
     
    dropoff
  </div>
  <div>
    <strong>
      booking.pickupDate
    </strong>
     
    date-time
  </div>
</div>
`;
